sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -438,6 +439,19 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("test concurrently generating collation keys") {
```

Comment:
Can this test go to `CollationFactorySuite`?

Reply:
it could but I decided to put it here because it would require adding a new dependency for parallel collections which I'd like to avoid